students/johnn/project/dcd.py

```diff
@@ -15,187 +15,443 @@
 import queue
 import socket
 
-# 
-parser = optparse.OptionParser()
-parser.add_option("-a", "--admin_port", dest="admin_port", help="Port of the admin interface on localhost")
-parser.add_option("-p", "--pub_port", dest="pub_port", help="Port of the publisher interface on localhost")
-parser.add_option("-d", "--debug", dest="debug", action="store_true", help="Print debug information to the screen")
 
-(options, args) = parser.parse_args()
+class Config():
+    """
+    create a common config object to hold common state
+    """
+    def __init__(self):
+        self.pub_queue = queue.Queue()
+        self.sub_queue = queue.Queue()
+        self.link_queue = queue.Queue()
+        self.data = {}
+        self.peers = {}
+    def set_value(self, key, value):
```

Comment:
why not let the user simply access the data attribute? or override __index__,and then the user could use conf[key] = value.